lib/omise/http_client.ex

```diff
@@ -0,0 +1,95 @@
+defmodule Omise.HTTPClient do
+  @moduledoc false
+
+  @base_vault_uri "https://vault.omise.co/"
+  @base_api_uri   "https://api.omise.co/"
+
+  @default_req_headers %{
+    "User-Agent"   => "OmiseElixir/#{Mix.Project.config[:version]} Elixir/#{System.version}",
+    "Content-type" => "application/json",
+  }
+  @default_http_options [timeout: :timer.seconds(60), recv_timeout: :timer.seconds(60)]
+
+  defmacro __using__(options) do
+    quote bind_quoted: [options: options] do
+      @endpoint Keyword.fetch!(options, :endpoint)
+      @key_type Keyword.get(options, :key_type, :secret_key)
+
+      def get(endpoint, query_params \\ [], options \\ []) do
+        Omise.HTTPClient.request(:get, @key_type, endpoint, [query_params: query_params], options)
+      end
+
+      def post(endpoint, body_params \\ [], options \\ []) do
+        Omise.HTTPClient.request(:post, @key_type, endpoint, [body_params: body_params], options)
+      end
+
+      def put(endpoint, body_params \\ [], options \\ []) do
+        Omise.HTTPClient.request(:put, @key_type, endpoint, [body_params: body_params], options)
+      end
+
+      def delete(endpoint, options \\ []) do
+        Omise.HTTPClient.request(:delete, @key_type, endpoint, [], options)
+      end
+    end
+  end
```

Comment:
Could be able extract `Omise.HTTPClient` module name to variable?

Reply:
Refactored as suggested here https://github.com/omise/omise-elixir/pull/19/commits/dc8cc812b52359658a62ee4d9877cb8052944bda